src/GraphQL/Validation/TypeInfo.cs

```diff
@@ -115,52 +115,52 @@ public void Enter(INode node, ValidationContext context)
                 return;
             }
 
-            if (node is Directive directive)
+            if (node is GraphQLDirective directive)
             {
-                _directive = _schema.Directives.Find(directive.Name);
+                _directive = _schema.Directives.Find((string)directive.Name); //TODO:!!!! alloc
```

Comment:
~40 places left with such comment. I'm working on it changin either to `ROM` or `.StringValue`